packages/tools/webpack/extract-depndencies-webpack-plugin.js

```diff
@@ -0,0 +1,127 @@
+// Inspired by "Dependency Extraction Webpack Plugin" by @wordpress team.
+// Link: https://github.com/WordPress/gutenberg/tree/trunk/packages/dependency-extraction-webpack-plugin
+
+const { sources: { RawSource } } = require( 'webpack' );
+
+module.exports = class ExtractDependenciesWebpackPlugin {
+	generateHandleName;
+	replaceDependencyNames;
+	generateAssetsFileName;
+
+	constructor( {
+		generateHandleName,
+		replaceDependencyNames,
+		generateAssetsFileName,
+	} = {} ) {
+		this.generateHandleName = generateHandleName || this.defaultGenerateHandleName;
+		this.replaceDependencyNames = replaceDependencyNames || this.defaultReplaceDependencyNames;
+		this.generateAssetsFileName = generateAssetsFileName || this.defaultGenerateAssetsFileName;
+	}
+
+	apply( compiler ) {
+		compiler.hooks.thisCompilation.tap( this.constructor.name, ( compilation ) => {
+			compilation.hooks.processAssets.tap( { name: this.constructor.name }, () => {
+				[ ...compilation.entrypoints ].forEach( ( [ id, entrypoint ] ) => {
+					const chunk = entrypoint.chunks.find( ( c ) => c.name === id );
+					const chunkJSFile = this.getFileFromChunk( chunk );
+
+					if ( ! chunkJSFile ) {
+						return;
+					}
+
+					const deps = this.getDepsFromChunk( compilation, chunk );
+
+					const assetFilename = this.generateAssetsFileName(
+						compilation.getPath( '[file]', { filename: chunkJSFile } )
+					);
+
+					const content = this.createAssetsFileContent(
+						id,
+						deps,
+						compilation.options.optimization.minimize
+					);
+
+					// Add source and file into compilation for webpack to output.
+					compilation.assets[ assetFilename ] = new RawSource( content );
+
+					chunk.files.add( assetFilename );
+				} );
+			} );
+		} );
+	}
+
+	getDepsFromChunk( compilation, chunk ) {
```

Comment:
Why don't we just take it from the package.json of each package?

Reply:
Firstly we think it will do the job, but when it will become a plugin that also third party will use I can't know if all their packages lives with one package.json or with multiple.
And it was easy enough to do so in this way.